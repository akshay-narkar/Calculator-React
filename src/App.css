.container {
  width: 600px;
  border: 1px solid rgb(0, 0, 0);
}

.display {
  font-size: 40px;
  background-color: #777;
  display: flex;
  height: 100px;
  color: white;
  align-items: center;
  justify-content: flex-end;
}

.op-btn {
  border: 2px solid rgb(197, 197, 197);

  font-size: 40px;
  width: 25%;
  background-color: #fd8d0e;
}
.btn {
  border: 2px solid rgb(197, 197, 197);
  font-size: 40px;
  width: 25%;
  background-color: #dddd;
}
.zero {
  font-size: 40px;
  width: 50%;
}

.row {
  height: 85px;
  display: flex;
}
