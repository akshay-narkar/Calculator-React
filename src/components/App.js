import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = (buttonName) => {
    const dataObject = { ...this.state };
    if (dataObject.total === 'error') {
      this.setState(() => ({
        total: null,
        next: null,
        operation: null,
      }));
      return;
    }
    const { next, operation, total } = calculate(dataObject, buttonName);
    this.setState(() => (
      {
        next,
        total,
        operation,
      }
    ));
  }

  render() {
    const { total } = this.state;
    return (
      <div className="container">
        <Display result={total} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}
