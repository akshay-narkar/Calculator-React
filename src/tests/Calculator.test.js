import renderer from 'react-test-renderer';
import App from '../components/App';

it('UI is unaltered wrt first snapshot', () => {
  const elem = renderer.create(<App />).toJSON();

  expect(elem).toMatchSnapshot();
});
