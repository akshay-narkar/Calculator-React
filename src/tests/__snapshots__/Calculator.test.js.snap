// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is unaltered wrt first snapshot 1`] = `
<div
  className="calcclass"
>
  <h5
    className="calclefttext"
  >
    Let’s do some math!
  </h5>
  <div
    className="container"
  >
    <div
      className="display"
    >
      <span
        id="text"
      />
    </div>
    <div
      className="button-panel"
    >
      <div
        className="row"
      >
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#fd8d0e",
              "width": "25%",
            }
          }
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#fd8d0e",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#fd8d0e",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#fd8d0e",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ddd",
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="btn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#fd8d0e",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
